neurom/core/types.py

```diff
@@ -64,6 +81,39 @@ class NeuriteType(IntEnum):
     custom9 = SectionType.custom9
     custom10 = SectionType.custom10
 
+    axon_carrying_dendrite = CompositeType(
+        value=1000,
+        types=(SectionType.basal_dendrite, SectionType.axon),
+    )
+
+    def __eq__(self, other):
+        """Logic for checking single and composite types."""
+        is_self_composite = hasattr(self, "value") and isinstance(self.value, CompositeType)
+        is_other_composite = hasattr(other, "value") and isinstance(other.value, CompositeType)
+
+        if is_self_composite:
+            if is_other_composite:
+                return self.value.subtypes == other.value.subtypes
+            return other in self.value.subtypes
+
+        if is_other_composite:
+            return self in other.value.subtypes
+
+        return super().__eq__(other)
+
+    def __hash__(self):
+        return self.value
+
+    def __new__(cls, value):
+        """Create a new integer enum entry and replace its `_value_` entry with the value object.
+
+        This allows passing the CompositeType instance to self.value.
+        """
+        obj = int.__new__(cls, value)
+        if isinstance(value, CompositeType):
+            obj._value_ = value
```

Comment:
Isn't it dangerous? Can `obj` still perform all `int` operations after this change?

Reply:
I wouldn't say it's dangerous because it's part of the documentation:
https://docs.python.org/3/library/enum.html#finer-points

And yes value is either a regular integer or a CompositeType, which is also of integer type. If `_value_` is not assigned, then it keeps only the `int(value)` instead of the CompositeType and its subtypes.